### ISSUE TITLE
registerSecret - fail instead of returning false

### ISSUE THREAD

**Issue by loredanacirstea** [raiden-network/spec#76]
Currently, `SecretRegistry.registerSecret` returns `false` but does not fail if the secret is already registered.
I would rather make this `revert` clearly than return `false`. I would also remove the `TokenNetwork.registerSecret` wrapper which doesn't do anything else but call the above function. 

`registerSecret` is called directly by the Raiden client and it is not used by another contract - so no reason to return a value (only useful inside EVM). Checking if the transaction failed should be the way to determine if the secret was registered; other methods are not that straightforward (e.g.`SecretRevealed` event)

**Comment by LefterisJP** [comment]
Yes after decoupling `unlock` and `registerSecret` there is no reason to return `false` anymore.

> you can also add error messages now.

Tools still need a bit of time to implement reading this message though. Remix just merged the reading into its alpha version.
